lib/widgets/content.dart

```diff
@@ -248,25 +248,274 @@ class MessageImage extends StatelessWidget {
   }
 }
 
-class CodeBlock extends StatelessWidget {
-  const CodeBlock({super.key, required this.node});
+Widget _buildCodeBlock({
+  required CodeBlockNode node,
+}) {
+  if (node.spans.isEmpty) {
+    return const CodeBlockBasic(text: '');
+  } else if (node.spans.length == 1 && node.spans.first.className == ''){
+    return CodeBlockBasic(text: node.spans.first.text);
+  } else {
+    return CodeBlockHighlighted(node: node);
```

Comment:
Can we unify the different cases here, and have just one CodeBlock widget that behaves like this CodeBlockHighlighted?

Or perhaps with a `_CodeBlockContainer` widget if that helps with the code organization — that one is fine, but what I really mean here is that ideally we'd have one `CodeBlock` instead of `_buildCodeBlock`, `CodeBlockBasic`, and `CodeBlockHighlighted`.

I know this resembles `_buildBlockInlineContainer` and its helpers. Those set a bit of a bad example, I think — I feel like they ended up somewhat messy, but I was paranoid about performance there. Specifically, every paragraph (and some other things too) in every message is going to go through that code path, and most of them don't contain any links (which are the thing that makes `_BlockInlineContainer` complicated and stateful), and I was worried about making every paragraph into a StatefulWidget when most of them don't need any state.

Here, code blocks aren't nearly so ubiquitous as paragraphs are, so I don't feel such a need to be paranoid; and in any case it's not clear that what CodeBlockHighlighted is doing (for a given number of spans) will be any more expensive than what CodeBlockBasic is doing. So we can cheerfully keep things simple.

Reply:
ack, unified them into single `CodeBlock` widget